packages/mui-material/src/Autocomplete/Autocomplete.js

```diff
@@ -552,7 +552,14 @@ const Autocomplete = React.forwardRef(function Autocomplete(inProps, ref) {
   );
 
   const renderGroup = renderGroupProp || defaultRenderGroup;
-  const defaultRenderOption = (props2, option) => <li {...props2}>{getOptionLabel(option)}</li>;
+  const defaultRenderOption = (props2, option) => {
+    const { key, ...otherProps } = props2;
+    return (
+      <li key={key} {...otherProps}>
+        {getOptionLabel(option)}
+      </li>
+    );
+  };
```

Comment:
The spread is not a very fast operator. Wouldn't this be better (under @romgrk control, who I think did changes like this in the MUI X codebase to improve performance)?

``` suggestion
  const defaultRenderOption = (props2, option) => {
    // Need to clearly apply key because of https://github.com/vercel/next.js/issues/55642
    return (
      <li key={props2.key} {...props2}>
        {getOptionLabel(option)}
      </li>
    );
  };
```

Reply:
Handled in #40968